src/apis/brain.js

```diff
@@ -409,73 +411,119 @@ export const createAsset = async params =>
+            let counterCallsToInternalActions = 0;
+
+            // we need to perform a few actions before declaring the listing of the asset as successful
+            const performInternalAction = () => {
+              counterCallsToInternalActions++;
+              if(counterCallsToInternalActions === requiredCallsToInternalActions){
+                onSuccess(() => {
+                  updateNotification(id, {
+                    listAssetProps: {
+                      assetName: assetName,
+                      assetId: futureAssetId,
+                    },
+                    status: 'success',
+                  })
+                }, futureAssetId)
+
+                resolve(receipt.status);
               }
-              axios.post(S3_UPLOAD_URL,
-                data, {
-                  headers: {
-                    'Content-Type': 'multipart/form-data'
-                  }
-                }
-              ).then((response) => {
-                debug('success');
-              })
-              .catch((err) => {
-                debug('fail');
-              });
             }
 
-            onSuccess(() => {
-              axios.post(MYBIT_API_COLLATERAL, {
-                address: params.userAddress,
-                escrow: collateral,
-                assetId: futureAssetId,
-              }).catch((error) => {
-                debug(error);
-              })
-
-              updateNotification(id, {
-                listAssetProps: {
-                  assetName: assetName,
-                  assetId: futureAssetId,
-                },
-                status: 'success',
-              })
-            }, futureAssetId)
+            updateAirTableWithNewAsset(futureAssetId, assetName, country, city, collateralMyb, collateralPercentage, performInternalAction)
+            createEntryForNewCollateral(userAddress, collateral, futureAssetId, performInternalAction);
+            filesUploaded && uploadFilesToAWS(futureAssetId, fileList, performInternalAction);
+
           } else {
             updateNotification(id, {
               listAssetProps: {
                 assetName: assetName,
               },
               status: 'error',
             });
+            resolve(assetCreationResponse);
           }
-
-          debug(receipt)
-          resolve(receipt.status);
         });
 
-      resolve(assetCreationResponse);
     } catch (err) {
       reject(err);
     }
   });
 
+const uploadFilesToAWS = async (
+  assetId,
+  fileList,
+  performInternalAction,
+) => {
+  try{
+    let data = new FormData();
+    data.append('assetId', assetId);
+    for(const file of fileList){
+      data.append('file', file.originFileObj);
+    }
+    await axios.post(S3_UPLOAD_URL,
+      data, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      }
+    )
+
+    performInternalAction();
+  } catch(err){
+    setTimeout(() => uploadFilesToAWS(assetId, fileList, performInternalAction), 5000);
```

Comment:
Doesn't this just keep retrying? That seems like a super easy way to get a server banned.